tools/data-api/internal/repositories/services.go

```diff
@@ -1,227 +1,411 @@
 package repositories
 
 import (
+	"encoding/json"
 	"fmt"
-	"net/http"
-
-	"github.com/hashicorp/go-azure-helpers/lang/pointer"
+	"os"
+	"strings"
 )
 
 type ServicesRepository interface {
 	GetByName(serviceName string, serviceType ServiceType) (*ServiceDetails, error)
-
 	GetAll(serviceType ServiceType) (*[]ServiceDetails, error)
+	ClearCache() error
 }
 
 var _ ServicesRepository = &ServicesRepositoryImpl{}
 
 type ServicesRepositoryImpl struct {
-	// TODO:
-	// directory string
+	// directory is where the api definitions for all services exist - this is where the server will read from
+	directory string
+
+	// graphV1 is a map of service name to string
+	graphV1 *map[string]string
+
+	// graphV2 is a map of service name to string
+	graphV2 *map[string]string
+
+	// resourceManager is map of service name to ServiceDetails, the ServiceDetails contain all the
+	// Service, Version and Resource definitions loaded and unmarshalled from the JSON API definitions
+	resourceManager *map[string]ServiceDetails
+
+	// serviceNames is a list of services which the server will load
+	serviceNames *[]string
+}
+
+func NewServicesRepository(directory string, serviceType ServiceType, serviceNames *[]string) ServicesRepository {
+	return &ServicesRepositoryImpl{
+		directory:    fmt.Sprintf("%s%s", directory, serviceType),
+		serviceNames: serviceNames,
+	}
+}
+
+func (s *ServicesRepositoryImpl) ClearCache() error {
+	// TODO This gets automatically run if a file has changed and when serve-watch is used
+	// TODO this finds the files needed and loads it
+	// TODO file watching
+	s.resourceManager = nil
+
+	return nil
+}
+
+func (s *ServicesRepositoryImpl) GetAll(serviceType ServiceType) (*[]ServiceDetails, error) {
+	// GetAll calls GetByName for all the service names passed to the serve command, or for all the services available in the api definitions directory
+	serviceDetails := make([]ServiceDetails, 0)
+
+	if s.resourceManager == nil {
```

Comment:
for a follow up PR, but we're gonna want to add a lock when reloading this data/clearing the cache so that we only load this a single time - so we should add TODOs here